### PR TITLE
task: update golangci-lint to 1.39.0

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Updating our linting

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -39,7 +40,6 @@ linters:
     - gosimple
     - govet
     - ineffassign
-    - interfacer
```
Comment (gssbzn) [inline .golangci.yml:42]: deprecated rule

File: .golangci.yml [hunk 2]
```diff
@@ -39,12 +40,10 @@ linters:
...
     - misspell
     - nakedret
     - nolintlint
     - rowserrcheck
-    - scopelint
```
Comment (gssbzn) [inline .golangci.yml:47]: deprecated and replaced by `exportloopref`

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by gssbzn** [comment]
@themantissa this is a minor update so I'll just merge

**Pull request merged**